src/main/java/seedu/address/ui/DeadlineTaskListPane.java

```diff
@@ -17,10 +23,27 @@
     @FXML
     private ListView<IndexedItem<DeadlineTask>> deadlineTaskListView;
 
-    public DeadlineTaskListPane(ObservableList<IndexedItem<DeadlineTask>> deadlineTaskList) {
+    @FXML
+    private Label listedDeadlineCounter;
+    @FXML
+    private Label todayUnfinishedDeadlineCounter;
+    @FXML
+    private Label overdueDeadlineCounter;
+    @FXML
+    private Label unfinishedDeadlineCounter;
+
+    public DeadlineTaskListPane(ReadOnlyModel model) {
         super(FXML);
-        deadlineTaskListView.setItems(deadlineTaskList);
+        deadlineTaskListView.setItems(model.getDeadlineTaskList());
         deadlineTaskListView.setCellFactory(listView -> new DeadlineTaskListCell());
+        listedDeadlineCounter.textProperty().bind(Bindings.size(model.getDeadlineTaskList())
+                                                  .asString("Number of Deadlines listed: %d"));
+        todayUnfinishedDeadlineCounter.textProperty().bind(Bindings.size(model.getDeadlineTaskList(new TaskWillHappenTodayPredicate(LocalDateTime.now())))
```

Comment:
One major problem though: the `TaskWillHappenTodayPredicate` will always compare with the time at which it was created. In order words: the counter will not change with time.